### PR TITLE
metadata-service: add `connectorTestSuitesOptions`

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/7548

We want to introduce new metadata fields for connectors to:
* Configure which test suite will run in CI
* Explicitely map test credentials in secret store to a connector test suite

## How
* Introduce a new top level `connectorTestSuitesOptions` to configure test suite runs.

## How will this be used?

This will help in conditionally running specific test suites according to the content of the connector metadata file.

### Examples
* A source Python  connector on which we want to run `unitTests`,  `integrationTests` (without secrets) and `acceptanceTests` (with secrets):
```yaml
# in metadata.yaml
connectorTestSuitesOptions:
  - suite: unitTests
  - suite: integrationTests
  - suite: acceptanceTests
    testSecrets:
      - name: SECRET_FOR_MY_AWESOME_SOURCE # The actual secret name in the secret store
        fileName: config.yaml # name of the file airbyte-ci will create after pulling the secret from the secret store
        secretStore:
          type: GSM
          alias: airbyte-internal-secret-store # This is to hide the actual address of the secret store, a mapping between alias and address will be stored as a secret value in GHA
 ```

* A source java connector on which we want to run unit, integration and acceptance tests without secrets:
```yaml
connectorTestSuitesOptions:
  - suite: unitTests
  - suite: integrationTests
  - suite: acceptanceTests
 ```

## User Impact
None as these new fields  are not used ATM.

## Follow up work
* https://github.com/airbytehq/airbyte-internal-issues/issues/7549
* https://github.com/airbytehq/airbyte-internal-issues/issues/7551
* https://github.com/airbytehq/airbyte-internal-issues/issues/7552

### CODE REVIEW

**Comment by alafanechere** [comment]
* **#38116** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38116?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by bnchrch — COMMENTED** [review]
Just one thought on string enum vs model enum I'd love your thoughts on

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorTestSuiteOptions.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+required:
+  - suite
+additionalProperties: false
+properties:
+  suite:
+    description: "Name of the configured test suite"
+    type: string
+    enum:
+      - "unitTests"
+      - "integrationTests"
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorTestSuiteOptions.yaml:16]: @bnchrch  I think I like the `enabled` field. But I think I prefer keeping per test suites secrets as it explicitly declares test suite secrets requirements.
If we'd go with your suggestion `airbyte-ci` would have to  eagerly mount the secrets declared in settings to each test suites even if they're not required. I'd prefer to keep things as explicit as possible and have a way to assess which test suite needs which secret... Does it sound reasonable?

[the remark this answers is not in the document]
Comment (bnchrch) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorTestSuiteOptions.yaml:16]: Sounds very reasonable! Approved!

**Review by bnchrch — APPROVED** [review]
Added via Giphy

**Pull request merged**